homeassistant/components/aussie_broadband/sensor.py

```diff
@@ -0,0 +1,193 @@
+"""Support for Aussie Broadband metric sensors."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from homeassistant.components.sensor import STATE_CLASS_TOTAL_INCREASING, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_SCAN_INTERVAL, DATA_KILOBYTES, DATA_MEGABYTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DEFAULT_UPDATE_INTERVAL, DOMAIN, SERVICE_ID
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+):
+    """Set up the Aussie Broadband sensor platform from a config entry."""
+    client = hass.data[DOMAIN][entry.entry_id]["client"]
+    services = hass.data[DOMAIN][entry.entry_id]["services"]
+    update_interval = timedelta(
+        minutes=entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_UPDATE_INTERVAL)
+    )
+
+    # Create an appropriate refresh function
+    def update_data_factory(service_id):
+        async def async_update_data():
+            return await client.get_usage(service_id)
+
+        return async_update_data
+
+    entities = []
+    for service in services:
```

Comment:
Move the coordinator construction to the init file, this can be shared between platforms

Reply:
This has now been implemented.